### DISCUSSION TITLE
Upgrading; do I need to remove OpenVPN 2.x before installing version 3?

### DISCUSSION THREAD

**Discussion by dsgiigh** [OpenVPN/openvpn3-linux#122] [Q&A - Our support arena] [answered]
I'm running the client (2.4.7) on my Ubuntu system, and would like to install the new version. Do I need to fully remove v2 first? Or will "sudo apt install openvpn3" take care of it?

**Comment by dsommers** [comment, marked as answer]
No, the `openvpn3` package and `openvpn` package can be installed in parallel, by design.

The OpenVPN 2.5.7 ships a single binary, `/usr/sbin/openvpn`.

OpenVPN 3 Linux ships more executables:
* `openvpn3` - the main user interface, also for unprivileged (non-root) users
* `openvpn3-admin` - a tool where most of the functionality requires `root` privileges
* `openvpn2` - An OpenVPN 3 wrapper simulating the old `openvpn` command line, but only with client options supported currently.  What is different from the `openvpn` approach is that `/usr/bin/openvpn2` can be run by unprivileged users.

OpenVPN 3 Linux also ships with `openvpn3-as`, which is used to download the configuration to connect to an OpenVPN Access Server.

**Reply by dsgiigh** [reply]
Thanks very much for the info. I installed it and it seemed to be working via the command line (openvpn3 session-start --config AS:ovpn.sonic.net). However, when ran "openvpn --version" the result was still 2.4.7 -- so I'm apparently missing something.

**Reply by dsommers** [reply]
You are mixing up the `openvpn` command and the `openvpn3` command.

On my own system:
```
$ openvpn --version
OpenVPN 2.5.6 x86_64-redhat-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Mar 16 2022
library versions: OpenSSL 1.1.1k  FIPS 25 Mar 2021, LZO 2.08
Originally developed by James Yonan
Copyright (C) 2002-2022 OpenVPN Inc <sales@openvpn.net>
Compile time defines: enable_async_push=yes enable_comp_stub=no enable_crypto_ofb_cfb=yes enable_debug=yes enable_def_auth=yes enable_dependency_tracking=no enable_dlopen=unknown enable_dlopen_self=unknown enable_dlopen_self_static=unknown enable_fast_install=yes enable_fragment=yes enable_iproute2=no enable_libtool_lock=yes enable_lz4=yes enable_lzo=yes enable_management=yes enable_multihome=yes enable_pam_dlopen=no enable_pedantic=no enable_pf=yes enable_pkcs11=yes enable_plugin_auth_pam=yes enable_plugin_down_root=yes enable_plugins=yes enable_port_share=yes enable_selinux=yes enable_shared=yes enable_shared_with_static_runtimes=no enable_silent_rules=yes enable_small=no enable_static=yes enable_strict=no enable_strict_options=no enable_systemd=yes enable_werror=no enable_win32_dll=yes enable_x509_alt_username=yes with_aix_soname=aix with_crypto_library=openssl with_gnu_ld=yes with_mem_check=no with_sysroot=no
$ which openvpn
/usr/sbin/openvpn

$ openvpn3 --version
openvpn3: Unknown command '--version'
$ openvpn3 version
OpenVPN 3/Linux git:next-master:a523112cb5f81bd7+ (openvpn3)
OpenVPN core 3.git:HEAD:46b13ec6 linux x86_64 64-bit built on May 20 2022 16:45:51
Copyright (C) 2012-2020 OpenVPN Inc. All rights reserved.
$ which openvpn3
/usr/bin/openvpn3
```

The first `openvpn --version` is from the `openvpn` package.  The `openvpn3` command comes from the `openvpn3` package.
